zkml/src/layers/transformer/softmax.rs

```diff
@@ -7,15 +9,29 @@ use crate::{
 
 #[derive(Debug, Clone)]
 pub struct Softmax<N> {
+    // By default, it's equal to 1
     pub scale: N,
+    // By default, softmax is going to be applied on the full tensor.
+    // You can specificy a dimen to apply softmax on. For example, for a tensor  of shape [2,3,4],
+    // if apply_on_dim = 2, then softmax will be applied on every chunks of 3 * 4 elements each.
```

Comment:
Shoudln't this be `apply_on_dim = 0`?

Reply:
Changed the definition of dimension, much clearer and i actually fixed a buggy test with that.